.travis.yml

```diff
@@ -17,16 +17,28 @@ cache:
   cargo: true
 before_script:
   - (which cargo-install-update && cargo install-update cargo-update) || cargo install cargo-update
-  - (which rustfmt && cargo install-update rustfmt) || cargo install rustfmt
   - (which cargo-prune && cargo install-update cargo-prune) || cargo install cargo-prune
+  - if [ "${TRAVIS_RUST_VERSION}" = stable ]; then
+      (which rustfmt && cargo install-update rustfmt) || cargo install rustfmt;
+    elif [ "${TRAVIS_OS_NAME}" = linux ]; then
+      clippy_vers=0.0.104;
+      if ! cargo clippy --version | grep -q $clippy_vers; then
+        cargo install clippy --vers=$clippy_vers --force;
+      fi
+    fi
 script:
   - if [ "${TRAVIS_RUST_VERSION}" = stable ]; then
         (
             set -x;
             cargo fmt -- --write-mode=diff &&
             cargo test --verbose --release
         );
+    elif [ "${TRAVIS_OS_NAME}" = linux ]; then
+        (
+            set -x;
+            cargo clippy --features=cargo-clippy &&
```

Comment:
Do we need to provide the features=cargo-clippy ? Routing travis doesn't seem to.